src/recommendationservice/metrics.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/python
+
+from gc import callbacks
+from typing import Iterable
+import psutil
+
+from opentelemetry.metrics import (
+    CallbackOptions,
+    Observation,
+)
+
+# RAM usage
+def ram_usage_callback(options: CallbackOptions) -> Iterable[Observation]:
+    observations = []    
+    ram_percent = psutil.virtual_memory().percent
+    print(f"ram_percent: {ram_percent}")
+    labels = {"dimension": "value"}
+    observations.append(Observation(ram_percent, labels))
+    
+    return observations
+
+# CPU Time callback
+def cpu_time_callback(options: CallbackOptions) -> Iterable[Observation]:
+    observations = []
+    with open("/proc/stat") as procstat:
+        procstat.readline()  # skip the first line
+        for line in procstat:
+            if not line.startswith("cpu"): break
+            cpu, *states = line.split()
+            observations.append(Observation(int(states[0]) // 100, {"cpu": cpu, "state": "user"}))
+            observations.append(Observation(int(states[1]) // 100, {"cpu": cpu, "state": "nice"}))
+            observations.append(Observation(int(states[2]) // 100, {"cpu": cpu, "state": "system"}))
+            # ... other states
+    return observations
+
+# CPU usage
+def cpu_usage_callback(options: CallbackOptions) -> Iterable[Observation]:
+    observations = []    
+    for (number, percent) in enumerate(psutil.cpu_percent(percpu=True)):
+        print(f"cpu_number: {number}, cpu_percent: {percent}")
+        labels = {"cpu_number": str(number)}
+        observations.append(Observation(percent, labels))
+        
+    return observations
+        
+def observable_counter_callback(options: CallbackOptions) -> Iterable[Observation]:
+    yield Observation(1, {})
+
+
+def observable_up_down_counter_callback(
+    options: CallbackOptions,
+) -> Iterable[Observation]:
+    yield Observation(-10, {})
+
+
+def observable_gauge_callback(options: CallbackOptions) -> Iterable[Observation]:
+    yield Observation(9, {})
+
+
+def init_metrics(meter):
+
+    # Requests counter
+    list_recommendations_request_counter = meter.create_counter(
```

Comment:
To align with the other manual counter, I'd change the name to
`app_recommendation_requests_counter`
and
`app.recommendation.requests.counter`
(but I'm fine with the current logic as well)

Reply:
Yeah, I've noticed the same.

Reply:
I changed the names of both metrics to use "_" instead of "."

Reply:
I'm also suggesting to change "list_" -> "app_".
When we have all the services implemented with manual metrics, it becomes easy to find them with consistent naming = all manually created metrics start with "app_". We agreed this in the last SIG meeting.

Reply:
Moot point now, since [@fatsheep9146 recommended that metric be removed](https://github.com/open-telemetry/opentelemetry-demo/pull/432#discussion_r994135299), since that should be handled by grace auto-instrumentation metrics (which are not yet in place).